### DISCUSSION TITLE
Uno Material and .net 6

### DISCUSSION THREAD

**Discussion by iot-crazy** [unoplatform/uno#8528] [Q&A] [answered]
Hi All,

I'm trying to follow the Uno Material tutorial here : [https://platform.uno/docs/articles/guides/uno-material-walkthrough.html?tabs=tabid-vswin](https://platform.uno/docs/articles/guides/uno-material-walkthrough.html?tabs=tabid-vswin) but having some problems.

Noting that some things have been moved to Uno.Toolkit.UI  I've tried to get the example code working but have struggled to get the right combination.

The code will build, but I get an exception about different styles not being available, such as MaterialDividerStyle.

**Comment by jeromelaban** [comment]
Thanks. Can you create a sample and post the zip file here?

**Comment by iot-crazy** [comment]
Sorry, forgot to post the URL to the repo:
https://github.com/iot-crazy/UnoMaterialTryout

**Comment by iot-crazy** [comment]
I also think I have it working now by using these lines in the app.xaml

```
<ToolkitResources xmlns="using:Uno.Toolkit.UI"/>
<MaterialResources xmlns="using:Uno.Material"/>
<MaterialToolkitColors xmlns="using:Uno.Toolkit.UI.Material" />
<MaterialToolkitResources xmlns="using:Uno.Toolkit.UI.Material" />
```

**Reply by kazo0** [reply]
FYI, the `MaterialToolkitColors` class no longer exists in newer versions of the Uno.Toolkit.UI.Material library, it relies directly on the `MaterialColors` class coming from `Uno.Material`.

**Comment by kazo0** [comment, marked as answer]
The documentation for Uno.Material is currently being updated to reflect the changes described below, see https://github.com/unoplatform/uno/issues/8475

NuGet Packages you will need to have installed:
- [Uno.Material](https://www.nuget.org/packages/Uno.Material/)
- [Uno.Toolkit.UI.Material](https://www.nuget.org/packages/Uno.Toolkit.UI.Material/) (has dependency on [Uno.Toolkit.UI](https://www.nuget.org/packages/Uno.Toolkit.UI/))

⚠️ Note: If you are building a WinAppSDK project then you will need to install the WinUI versions of the above two packages:
- [Uno.Material.WinUI](https://www.nuget.org/packages/Uno.Material.WinUI/)
- [Uno.Toolkit.WinUI.Material](https://www.nuget.org/packages/Uno.Toolkit.WinUI.Material/) (has dependency on [Uno.Toolkit.WinUI](https://www.nuget.org/packages/Uno.Toolkit.WinUI/))

The resource initialization within your `App.xaml` will have to be done in a specific order, it should look something like this:

```xml
<Application.Resources>
	<ResourceDictionary>
		<ResourceDictionary.MergedDictionaries>
			<!--  Load WinUI resources  -->
			<XamlControlsResources xmlns="using:Microsoft.UI.Xaml.Controls" />

			<!--  Load Uno.UI.Toolkit resources  -->
			<ToolkitResources xmlns="using:Uno.Toolkit.UI" />

			<!--  Load Material resources  -->
			<MaterialColors xmlns="using:Uno.Material" />
			<MaterialResources xmlns="using:Uno.Material" />

			<!--  Load Material Toolkit resources  -->
			<MaterialToolkitResources xmlns="using:Uno.Toolkit.UI.Material" />

			<!--  Load remaining application resources  -->
		</ResourceDictionary.MergedDictionaries>
	</ResourceDictionary>
</Application.Resources>
```
